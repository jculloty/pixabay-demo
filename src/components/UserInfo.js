import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import withApi from './ApiContext';

class UserInfo extends PureComponent {
  static propTypes = {
    user: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
  }

  queryUser = () => {
    this.props.queryApi(`user:${this.props.user.name}`);
    this.props.history.push('/');
  }

  render() {
    const { user } = this.props;

    return (
      <div onClick={this.queryUser}>
        <img src={user.url} alt="" className="user-image" />
        <span style={{ fontSize: '32px', paddingTop: '25px', textAlign: 'center' }}>{user.name}</span>
      </div>
    );
  }
}

export default withRouter(withApi(UserInfo, ['queryApi']));
