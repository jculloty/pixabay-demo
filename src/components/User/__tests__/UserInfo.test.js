import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import UserInfo from '../UserInfo';

const buildComponent = function(user = { name: 'John-Doe', total: 100, url: 'www.jd.com', id: 99 }) {
  return (
    <MemoryRouter>
      <UserInfo user={user} />
    </MemoryRouter>
  );
};

describe('User: UserInfo', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Write more tests', () => {
    expect(true).toBe(false);
  });
});
