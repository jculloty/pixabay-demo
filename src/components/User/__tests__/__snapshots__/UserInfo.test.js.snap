// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User: UserInfo renders correctly 1`] = `
<a
  href="/user/99-John-Doe"
  onClick={[Function]}
>
  <img
    alt=""
    className="user-image"
    src="www.jd.com"
  />
  <span
    style={
      Object {
        "fontSize": "32px",
        "paddingTop": "25px",
        "textAlign": "center",
      }
    }
  >
    John-Doe
  </span>
</a>
`;
